Add base class

SrdjanStevanetic opened the pull request.

SrdjanStevanetic committed:
1) added base class; 2) started new translation evaluation module; 3)
retry for the wikidata service call

SrdjanStevanetic committed:
added bert topic html files

SrdjanStevanetic committed:
throw exception when wikidata can not be fetched from remote

SrdjanStevanetic committed:
new ner algorithm and some code improvements (removed interfaces, etc.)

SrdjanStevanetic committed:
service for fetching the translation evaluation collections

gsergiu commented on enrichment-common/src/main/java/eu/europeana/enrichment/common/commons/HelperFunctions.java:
```diff
@@ -396,4 +401,32 @@ public static String[] convertListToArray(List<String> itemList) {
 		return itemArr;
 	}
 
+	public static List<String> sortWikiIds(List<String> wikiIds) {
```
this should not be made, at it overules the relevance of the wikidata search results

gsergiu replied:
it is not part of the new algorithm specifications

SrdjanStevanetic replied:
I noticed the dbpedia does not sort it properly, and for the wiki search it is ok. I removed it for both.

gsergiu commented on enrichment-mongo/src/main/java/eu/europeana/enrichment/mongo/dao/ItemEntityDaoImpl.java:
```diff
@@ -28,48 +26,26 @@ public class ItemEntityDaoImpl implements ItemEntityDao{
-			}
-			return tmpResult;
-		}
+	public List<ItemEntityImpl> findAllItemEntities() {
```
this method should be removed as we don't want to load million of records into the memory

SrdjanStevanetic replied:
Improved for all entities.

gsergiu commented on enrichment-mongo/src/main/java/eu/europeana/enrichment/mongo/dao/NamedEntityAnnotationDao.java:
```diff
@@ -2,19 +2,18 @@
-	public NamedEntityAnnotation findNamedEntityAnnotationWithStoryIdItemIdAndWikidataId(String storyId, String itemId, String wikidataId);
-	public List<NamedEntityAnnotation> findNamedEntityAnnotation(String storyId, String itemId, String property, List<String> nerTools);
-	public void saveNamedEntityAnnotation(NamedEntityAnnotation entity);
+	public NamedEntityAnnotationImpl findNamedEntityAnnotation(String id);
```
the interfaces should still use the interfaces for the data model, the cast can be performed where needed

gsergiu replied:
will revert back later

gsergiu commented on enrichment-mongo/src/main/java/eu/europeana/enrichment/mongo/dao/NamedEntityDaoImpl.java:
```diff
@@ -78,53 +110,49 @@ public List<NamedEntityImpl> findAllNamedEntitiesByLabelAndType(String label, St
-			.filter(filters.toArray(Filter[]::new))
-            .iterator()
-			.toList();
+	public List<NamedEntityImpl> findNamedEntitiesWithAdditionalInformation(String storyId, String itemId, String fieldUserForNER, List<String> nerTools, boolean matchNerToolsExactly) {
```
this method has ca 150 lines of code, it must be refactored in smaller methods

gsergiu commented on enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/NERController.java:
```diff
@@ -78,31 +79,39 @@ public ResponseEntity<String> createNamedEntitiesStory(
-		}
-		else {	
-			enrichmentNerService.createNamedEntitiesForStory(storyId, property, nerToolsList, true, linkingList, translationTool, original, false);
+		if(force) {
```
this should overwrite the previously generated data, which means the that previously generated enrichments should be deleted

gsergiu replied:
should probably be allowed for admins, but that is to be implemented later

gsergiu commented on enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/NERController.java:
```diff
@@ -78,31 +79,39 @@ public ResponseEntity<String> createNamedEntitiesStory(
-			enrichmentNerService.createNamedEntitiesForStory(storyId, property, nerToolsList, true, linkingList, translationTool, original, false);
+		if(force) {
+			enrichmentNerService.createNamedEntitiesForStory(storyId, property, nerToolsList, linkingList, translationTool, original, false, true);
 			resultJsonLd="{\"Result\":\"The NER analysis has been successfully executed.\"}";
```
the result must include the results of the enrichment

gsergiu replied:
probably also return status 201

gsergiu commented on enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/NERController.java:
```diff
@@ -78,31 +79,39 @@ public ResponseEntity<String> createNamedEntitiesStory(
+			}
+			else {	
+				enrichmentNerService.createNamedEntitiesForStory(storyId, property, nerToolsList, linkingList, translationTool, original, false, true);
+				resultJsonLd="{\"Result\":\"The NER analysis has been successfully executed.\"}";
```
the result must include the results of the enrichment

gsergiu commented on enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/NERController.java:
```diff
@@ -159,31 +168,39 @@ public ResponseEntity<String> createNamedEntitiesItem(
-			resultJsonLd="{\"Result\" : \"NamedEntity-s have been already crrated! Please use get method to get them!\"}";
+		if(force) {
+			enrichmentNerService.createNamedEntitiesForItem(storyId, itemId, property, nerToolsList, linkingList, translationTool, original, false, true);
+			resultJsonLd="{\"Result\":\"The NER analysis has been successfully executed.\"}";			
```
the result must include the results of the enrichment

gsergiu commented on enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/NERController.java:
```diff
@@ -159,31 +168,39 @@ public ResponseEntity<String> createNamedEntitiesItem(
+			}
+			else {	
+				enrichmentNerService.createNamedEntitiesForItem(storyId, itemId, property, nerToolsList, linkingList, translationTool, original, false, true);
+				resultJsonLd="{\"Result\":\"The NER analysis has been successfully executed.\"}";
```
the result must include the results of the enrichment

gsergiu commented on enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/NERController.java:
```diff
@@ -78,31 +79,39 @@ public ResponseEntity<String> createNamedEntitiesStory(
+		else {
+			List<NamedEntityImpl> result = enrichmentNerService.getEntities(storyId, null, property, nerToolsList);			
+			if(!result.isEmpty()) {
+				resultJsonLd="{\"Result\" : \"NamedEntity-s have been already crrated! Please use get method to get them!\"}";
```
should probably return error response with status 422

gsergiu commented on enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/NERController.java:
```diff
@@ -159,31 +168,39 @@ public ResponseEntity<String> createNamedEntitiesItem(
+		else {
+			List<NamedEntityImpl> result = enrichmentNerService.getEntities(storyId, itemId, property, nerToolsList);
+			if(!result.isEmpty()) {
+				resultJsonLd="{\"Result\" : \"NamedEntity-s have been already created! Please use get method to get them!\"}";
```
should probably return error response with status 422

gsergiu reviewed (commented):
the code can be merged after fixing the comments, the NERL algorithm still needs to be reviewed and validated

SrdjanStevanetic committed:
1) new translation evaluation module; 2) updated wikidata type check for
places; 3) eTranslation endpoint; 4) excluded
spring-boot-starter-logging from the dependencies

SrdjanStevanetic committed:
1) added eTranslation endpoint locally; 2) new body fields added for the
annotation creation; 3) translation evaluation added for all 3
translation services (google, eTranslation and Deepl); 4) parallel
linking; 5) removed methods for fetching all items or stories into
memory

SrdjanStevanetic committed:
1) api method for NER analysis updates the existing named and position
entities; 2) api method for the annotation creation only creates the
annotations without NER analysis; 3) added parallel annotation creation

SrdjanStevanetic committed:
1) fixed error message bean config; 2) mongo search of entities with
various fields in one method; 3) tested NER and annotation api methods
